### ISSUE TITLE
absolute_path_linter() warns on non-UTF8 strings

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2836]
As observed in {DBItest}:

https://github.com/r-dbi/DBItest/blob/5bc5fd126c24d403dfde2e2da464421f1c9bada9/R/spec-meta-bind-arrow-stream.R#L424

```r
lint("'M\\xfcller'\n", absolute_path_linter())
# ℹ No lints found.
# Warning message:
# In regexpr(pattern = pattern, data, perl = TRUE, ...) :
#   input string 1 is invalid UTF-8
```

**Comment by MichaelChirico** [comment]
The same applies to `nonportable_path_linter()` of course
